ports/metal-cpp/CMakeLists.txt

```
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 3.4)
+project(metal-cpp)
+include(GNUInstallDirs)
+
+add_library(metal INTERFACE)
```

> Step 3. Generate the implementation
metal-cpp is a header-only library. To generate the implementation, in one of your .cpp files add the following code:

Because generating the implementation must be done exactly once, we should do it inside this package and install a static library containing the single implementation object.

```suggestion
add_library(metal impl.cpp)
```

`impl.cpp` should have:
```
#define NS_PRIVATE_IMPLEMENTATION
#define CA_PRIVATE_IMPLEMENTATION
#define MTL_PRIVATE_IMPLEMENTATION
#include <Foundation/Foundation.hpp>
#include <Metal/Metal.hpp>
#include <QuartzCore/QuartzCore.hpp>
```